jerry-core/parser/js/js-parser-statm.c

```diff
@@ -1644,6 +1661,172 @@ parser_parse_continue_statement (parser_context_t *context_p) /**< context */
+
+  switch (context_p->token.type)
+  {
+    case LEXER_LEFT_BRACE:
+    {
+      lexer_next_token (context_p);
+      parser_module_parse_import_item_list (context_p, import_node_p);
+
+      if (context_p->token.type != LEXER_RIGHT_BRACE)
+      {
+        parser_raise_error (context_p, PARSER_ERR_RIGHT_PAREN_EXPECTED);
+      }
+
+      lexer_next_token (context_p);
+      break;
+    }
+
+    case LEXER_MULTIPLY:
+    case LEXER_LITERAL:
+    {
+      parser_module_parse_import_item_list (context_p, import_node_p);
+      break;
+    }
+
+    default:
+    {
+      parser_raise_error (context_p, PARSER_ERR_LEFT_PAREN_MULTIPLY_LITERAL_EXPECTED);
+    }
+  }
+
+  // from
+  if (context_p->token.type == LEXER_LITERAL
+      && !lexer_compare_raw_identifier_to_current (context_p, "from", 4))
+  {
+    parser_raise_error (context_p, PARSER_ERR_FROM_EXPECTED);
+  }
+
+  // module name
+  lexer_expect_object_literal_id (context_p, false);
+  import_node_p->script_path_length = (prop_length_t)(context_p->lit_object.literal_p->prop.length + 1);
+  import_node_p->script_path =
+  (uint8_t *) parser_malloc (context_p, import_node_p->script_path_length * sizeof (uint8_t));
+
+  memcpy (import_node_p->script_path,
+          context_p->lit_object.literal_p->u.char_p,
+          import_node_p->script_path_length);
+  import_node_p->script_path[import_node_p->script_path_length - 1] = '\0';
+  lexer_next_token (context_p);
+
+  parser_module_add_import_node_to_context (context_p, import_node_p);
+} /* parser_parse_import_statement */
+
+/**
+ * Parse export statement.
+ */
+static void
+parser_parse_export_statement (parser_context_t *context_p) /**< context */
+{
+  JERRY_ASSERT (context_p->token.type == LEXER_KEYW_EXPORT);
+  parser_module_context_init (context_p);
+
+  parser_module_node_t *export_node_p = parser_module_get_export_node (context_p);
+
+  default_keyword_found:
+  lexer_next_token (context_p);
+
+  switch (context_p->token.type)
+  {
+    case LEXER_LEFT_BRACE:
+    {
+      lexer_next_token (context_p);
+      parser_module_parse_export_item_list (context_p, export_node_p);
+
+      if (context_p->token.type != LEXER_RIGHT_BRACE)
+      {
+        parser_raise_error (context_p, PARSER_ERR_RIGHT_PAREN_EXPECTED);
+      }
+
+      lexer_next_token (context_p);
+      break;
+    }
+
+    case LEXER_KEYW_DEFAULT:
+    {
+      // TODO: This part is going to be implemented in the next part of the patch.
+      goto default_keyword_found;
+      break;
+    }
+
+    case LEXER_MULTIPLY:
+    case LEXER_LITERAL:
+    {
+      parser_module_parse_export_item_list (context_p, export_node_p);
+      break;
+    }
+
+    case LEXER_KEYW_FUNCTION:
+    {
+      lexer_literal_t *function_name_p = (lexer_literal_t *) parser_malloc (context_p, sizeof (lexer_literal_t));
+      parser_parse_function_statement (context_p, function_name_p);
+
+      lexer_literal_t *import_name_p = function_name_p;
+      lexer_literal_t *local_name_p = function_name_p;
```

Comment:
There is no need for the two local vars, just simly pass for the next method the `function_name_p`.